game/src/core/utils/Vector2.java

```diff
@@ -0,0 +1,42 @@
+package core.utils;
+
+/**
+ * A record representing a 2D vector with x and y Float components.
+ *
+ * <p>Provides utility methods for vector operations such as addition, subtraction, scaling,
+ * normalization, and rotation.
+ *
+ * @param x The x component of the vector.
+ * @param y The y component of the vector.
+ * @see Point
+ * @see core.level.utils.Coordinate Coordinate
+ */
+public record Vector2(float x, float y) implements IVec2 {
+
+  /**
+   * Create a new vector from another vector.
+   *
+   * @param other The other vector.
+   */
+  public Vector2(IVec2 other) {
```

Comment:
@Flamtky was hältst du von einer statischen methode stattdessen? sowas wie `public static Vector2 of(IVec2 other)`? 

ditto bei den anderen konstruktoren?

das würde beim aufruf das ständige `new Vector2(foo)` ersparen?

Reply:
Ich habe die Methode im Interface, nur noch nicht so oft verwendet. Ich würde aber die Aufrufe dann anpassen.
https://github.com/Dungeon-CampusMinden/Dungeon/blob/955a733909cdc9ba5bb98de582bd1fcb8ef0e8ff/game/src/core/utils/IVec2.java#L43-L52

Reply:
das hätte den charme, dass es ein interface ist und man im prinzip als nutzer weitestgehende freiheiten hat. 

aber es sieht zugegebenermaßen irgendwie schräg aus - konstanten im interface, record-definitionen im interface ...

Reply:
@Flamtky was denkst du dazu? (vorschlag zwei kommentare wieter oben, einmal,aufklappen)

vorteile: das interface enthält alles relevante und erlaubt dem nutzer das ableiten/hinzumischen. die im interface definierte implementierung als record verstopft nicht den package-tree, gleichzeitig kann man die defaultmethoden aus dem interface direkt nutzen ...

nachteile: es sieht einfach super schräg aus. interfaces mit so viel implementierung - eigentlich sollte ein interface ja nur einen typ samt operationen definieren und eine gewisse schlankheit mitbringen.

Reply:
Das gefällt mir sehr gut. Die Konstanten im Interface finde ich prinzipiell akzeptabel. Das Record ist zwar ungewöhnlich bzw gut versteckt, aber das stellt für mich kein Problem dar. Alternativ könnte man jedoch auch wieder eine separate `public record Vector2` verwenden, wie zuvor. Ich habe zudem schon eine Variante gesehen, bei der ähnlich wie in unserem Fall vorgegangen wurde, aber die hatten eine separate `SimpleVector2`-Klasse die lediglich das Interface implementiert.

Reply:
ja, das record ist etwas versteckt - aber ich sehe das eher als eine art interne implementierung des interfaces, die dem kunden z.b. in den `of`-methoden zurcükgegeben wird.

für mich reicht das. wenn der kunde mehr will, soll er sich das interface selbst implementieren :) 

man könnte ggf. sogar hingehen und das record/die implementierung noch weiter verstecken, indem man es in eine innere klasse überführt. bin aber nicht sicher, ob das in interfaces überhaupt erlaubt ist.

Reply:
`Vector2` ist doch schon eine innere record vom Interface? Oder meinst du etwas anderes?